### PR TITLE
Nextflow_schema.json updates

### LINKED ISSUE #502: Update nextflow.json
- [ ] All parameters have a good explanation at hand
- [ ] All parameters are sorted in an order that is intuitive and makes sense to the user

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
close #502 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -16,16 +16,15 @@
                     "type": "string",
                     "default": "mapping",
                     "fa_icon": "fas fa-play",
-                    "description": "Starting step.",
-                    "help_text": "Only one step.",
+                    "description": "Starting step",
+                    "help_text": "The pipeline starts from this step and then runs through the subsequent steps.\n\n",
```
Comment (maxulysse) [inline nextflow_schema.json:20]:
```suggestion
                    "help_text": "The pipeline starts from this step and then runs through the possible subsequent steps.\n\n",
```

**Comment by maxulysse** [comment]
What's happening with all the ascat params?

**Comment by FriederikeHanssen** [comment]
> What's happening with all the ascat params?

Reorganized them a bit, some are ref files but some are tool configurations. 
Also removed ascat_chromosomes, since you can't actually manipualte it form the command line. It is used like this currently in the modules.config:

```
            "chrom_names": meta.sex == 'XX' ? params.ascat_chromosomes : "c(1:22, 'X', 'Y')",
```

**Comment by maxulysse** [comment]
Can you update CHANGELOG?

File: nextflow_schema.json [hunk 2]
```diff
@@ -50,96 +49,110 @@
...
-                    "pattern": "^((ascat|cnvkit|controlfreec|deepvariant|dnascope|dnaseq|freebayes|haplotypecaller|manta|merge|mpileup|msisensorpro|mutect2|snpeff|strelka|tiddit|tnscope|vep)*,?)*$"
+                "split_fastq": {
+                    "type": "integer",
+                    "default": 0,
+                    "fa_icon": "fas fa-clock",
+                    "description": "Specify how many reads each split of a FastQ file contains. Set 0 to turn of splitting at all.",
```
Comment (maxulysse) [inline nextflow_schema.json:59]:
```suggestion
                    "description": "Specify how many reads each split of a FastQ file contains. Set 0 to turn off splitting at all.",
```

Comment (maxulysse) [inline nextflow_schema.json]: Wouldn't it make sense to have that one back in the `fastq_preprocessing` section?

Comment (FriederikeHanssen) [reply nextflow_schema.json]: I am a bit conflicted about it. On the one hand it does something to the fastq files, on the other hand  it is used for speed up but doesn't change the reads themselves.

File: nextflow_schema.json [hunk 3]
```diff
@@ -189,13 +197,14 @@
                     "fa_icon": "fas fa-puzzle-piece",
                     "enum": ["bwa-mem", "bwa-mem2", "dragmap"],
                     "description": "Specify aligner to be used to map reads to reference genome.",
-                    "help_text": "> **WARNING** Current indices for `bwa` in AWS iGenomes are not compatible with `bwa-mem2` and `dragmap`.\n> `Sarek` will build them automatically if not provided.\n\n> **WARNING** BWA-mem2 is in active development\n> Sarek might not be able to require the right amount of resources for it at the moment\n> We recommend to use pre-built indexes.",
+                    "help_text": "Parameter chooses the mapper used. We recommend using pre-build indices. If they are not part of the available `igenome.config` they can be provided with `--bwa` . \nIf none are provided,  `Sarek` will build them automatically.",
```
Comment (maxulysse) [inline nextflow_schema.json:200]:
```suggestion
                    "help_text": "Parameter chooses the mapper used. We recommend using pre-build indices. If they are not part of the available `igenome.config` they can be provided with `--bwa`, `--bwamem2` or `--dragmap`. \nIf none are provided,  `Sarek` will build them automatically.",
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -438,197 +476,238 @@
...
+                        "canFam3",
+                        "danRer10",
+                        "dm6",
+                        "equCab2",
+                        "galGal4",
+                        "panTro4",
+                        "rn6",
+                        "sacCer3",
+                        "susScr3"
+                    ]
```
Comment (maxulysse) [inline nextflow_schema.json:521]: we can have `null` as well when using `--igenomes_ignore`

**Comment by maxulysse** [comment]
I noticed that we sometimes have `\n\n` and sometimes `\n`.
What about putting more consistency there.

File: nextflow_schema.json (hunk 4 above)
Comment (FriederikeHanssen) [reply nextflow_schema.json:521]: 🤦‍♀️ right, can I add it here or remove the whole list?

File: nextflow_schema.json (hunk 2 above)
Comment (maxulysse) [reply nextflow_schema.json]: Makes sense.

File: nextflow_schema.json (hunk 4 above)
Comment (maxulysse) [reply nextflow_schema.json:521]: We have checks for that in the groovy libs, so I'd say we can remove the whole list.

**Comment by maxulysse** [comment]
You have a conflict in CHANGELOG

File: nextflow_schema.json (hunk 4 above)
Comment (FriederikeHanssen) [reply nextflow_schema.json:521]: ah ok awesome

**Review by maxulysse — APPROVED** [review]

**Pull request merged**